src/xrAICore/Components/problem_solver_inline.h

```diff
@@ -362,7 +362,7 @@ IC void CProblemSolverAbstract::solve()
     m_solution_changed = true;
     m_current_state.clear();
 
-    m_failed = !ai().graph_engine().search(*this, reverse_search ? target_state() : current_state(),
+    m_failed = !GEnv.AISpace->graph_engine().search(*this, reverse_search ? target_state() : current_state(),
```

Comment:
There's a problem with using `ai()` in this place??

Reply:
ai() is inline method for CAI_Space (even not in namespace), but this method solve() belongs of parent AISpaceBase
`class CAI_Space : public AISpaceBase`
Actually, I don’t know how to do it right. The `ai()` method is a transfer artifact from xrGame, probably is need to rewrite the `solve()` method, but I do not understand the logic of its operation.